core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java

```diff
@@ -269,22 +273,25 @@ public NegotiationResult declined(ClaimToken token, String negotiationId) {
      * @return the number of processed ContractNegotiations.
      */
     private int sendContractOffers() {
-        var processes = negotiationStore.nextForState(ContractNegotiationStates.REQUESTING.code(), batchSize);
+        var processes = negotiationStore.nextForState(REQUESTING.code(), batchSize);
 
         for (ContractNegotiation process : processes) {
             var offer = process.getLastContractOffer();
-            var response = sendOffer(offer, process, ContractOfferRequest.Type.INITIAL);
-            if (response.isCompletedExceptionally()) {
-                process.transitionRequesting();
-                monitor.debug(format("[Consumer] Failed to send contract offer with id %s. ContractNegotiation %s stays in state %s.",
-                        offer.getId(), process.getId(), ContractNegotiationStates.from(process.getState())));
-                continue;
-            }
-
-            process.transitionRequested();
-            monitor.debug(String.format("[Consumer] ContractNegotiation %s is now in state %s.",
-                    process.getId(), ContractNegotiationStates.from(process.getState())));
-            negotiationStore.save(process);
+            sendOffer(offer, process, ContractOfferRequest.Type.INITIAL)
+                    .whenComplete((response, throwable) -> {
+                        if (throwable == null) {
+                            process.transitionRequested();
+                            negotiationStore.save(process);
+                            monitor.debug(String.format("[Consumer] ContractNegotiation %s is now in state %s.",
+                                    process.getId(), ContractNegotiationStates.from(process.getState())));
+                        } else {
+                            process.transitionRequesting();
+                            negotiationStore.save(process);
```

Comment:
@ndr-brt Beware that you are deviating from the logic which is currently limplemented here.

In the current implementation, the `negotiationStore.save(process);` is not called in case of an exception, due to the `continue` statement. 

@juliapampus was it a miss in the current implem?

Reply:
yes, I added it by purpose. Without saving the process, no `stateCount` gets incremented. 
I think this was an oversight.

Reply:
> yes, I added it by purpose. Without saving the process, no `stateCount` gets incremented. I think this was an oversight.

Yeap seems correct. As soon as @juliapampus confirms, it should ready for merge.